### ISSUE TITLE
Vulkan failing to initialize sporadically on Windows 11 (dual GPUs)

### ISSUE THREAD

**Issue by FredrikNoren** [gfx-rs/wgpu#2523]
**Description**
Hi, I'm running hello-triangle on a new Windows 11 laptop with dual GPUs (nvidia and ryzen integrated), but it's _sometimes_ failing to create the adapter ("Failed to find an appropriate adapter"). The computer is completely new, the nvidia and ryzen gpu drivers are both updated. I also checked which adapters it's listing and it's giving me 0 adapters found (see log).

The strange thing is that it has worked at times; when I just got the laptop, and then later right after the drivers were installed (but currently it fails every time).

**Repro steps**
I'm just using the hello-triangle example, and I added this to list the adapters:
```rust
log::info!("All available adapters: [{:?}]", instance.enumerate_adapters(wgpu::Backends::PRIMARY).map(|x| format!("{:?}", x.get_info())).join(", "));
```

**Extra materials**
Here's the log output from the program:
```
[00:00:01.254] (571c) WARN   Unable to find layer: VK_LAYER_KHRONOS_validation
[00:00:01.301] (571c) INFO   Instance version: 0x4020b0
[00:00:01.301] (571c) INFO   Enabling debug utils
[00:00:01.301] (571c) INFO   Enabling device properties2
[00:00:01.302] (571c) ERROR  GENERAL [Loader Message (0x0)]
	setupLoaderTrampPhysDevs:  Failed during dispatch call of 'vkEnumeratePhysicalDevices' to lower layers or loader to get count.
[00:00:01.303] (571c) ERROR  	objects: (type: INSTANCE, hndl: 0x1481a3501f0, name: ?)
[00:00:01.303] (571c) ERROR  GENERAL [Loader Message (0x0)]
	setupLoaderTrampPhysDevs:  Failed during dispatch call of 'vkEnumeratePhysicalDevices' to lower layers or loader to get count.
[00:00:01.303] (571c) ERROR  	objects: (type: INSTANCE, hndl: 0x1481a3501f0, name: ?)
[00:00:01.304] (571c) INFO   All available adapters: [""]
[00:00:01.305] (571c) ERROR  GENERAL [Loader Message (0x0)]
	setupLoaderTrampPhysDevs:  Failed during dispatch call of 'vkEnumeratePhysicalDevices' to lower layers or loader to get count.
[00:00:01.305] (571c) ERROR  	objects: (type: INSTANCE, hndl: 0x1481a3501f0, name: ?)
[00:00:01.306] (571c) ERROR  GENERAL [Loader Message (0x0)]
	setupLoaderTrampPhysDevs:  Failed during dispatch call of 'vkEnumeratePhysicalDevices' to lower layers or loader to get count.
[00:00:01.306] (571c) ERROR  	objects: (type: INSTANCE, hndl: 0x1481a3501f0, name: ?)
[00:00:01.307] (571c) INFO   Dropping Global
```

**Platform**
wgpu: 0.12.0, Windows 11, Nvidia RTX3070

**Comment by kvark** [comment]
Interesting! Looks like an issue with Vulkan Loader, but we should investigate and see if anything can be done to help it.

**Comment by FredrikNoren** [comment]
I tried running https://github.com/skeeto/vulkan-test and it tells me Vulkan loads fine. I also tried starting a game and that also worked (which used DirectX though).

**Comment by FredrikNoren** [comment]
Some additional info: This laptop had an external table/drawing screen connected to it (it's not actually my laptop but our graphical artists laptop), and we got it working by disconnecting that now. We tried that before but at that point it didn't work so not sure what changed now.

**Comment by FredrikNoren** [comment]
Another update: Disconnecting the wacom screen doesn't seem to help any more. We got it working from time to time by installing drivers, but then it will stop working again, and currently it doesn't seem to help any more. Also tried re-installing windows but that didn't help.

**Comment by FredrikNoren** [comment]
Another update: We bought another laptop of the same make, but we have the same problem with this one and it doesn't have any external devices connected, so that seems like that was unrelated. vulkan-test still reports it can create a device on this new machine, but wgpu reports 0 devices found.

**Comment by FredrikNoren** [comment]
I got a bit more information in the logs now: (I think it's because I'm building the release build with vulkan debug logging enabled now)

```
wgpu_hal::vulkan::instance
GENERAL [Loader Message (0x0)]
	vkEnumeratePhysicalDevices: One or more layers modified physical devices!Count returned by ICDs = 2, count returned above layers = 0
Info
09:52:21
wgpu_hal::vulkan::instance
	objects: (type: INSTANCE, hndl: 0x189d2958ba0, name: ?)
Info
09:52:21
wgpu_hal::vulkan::instance
GENERAL [Loader Message (0x0)]
	vkEnumeratePhysicalDevices: One or more layers modified physical devices!Count returned by ICDs = 2, count returned above layers = 0
Info
09:52:21
wgpu_hal::vulkan::instance
	objects: (type: INSTANCE, hndl: 0x189d2958ba0, name: ?)
Info
09:52:21
wgpu_hal::vulkan::instance
GENERAL [Loader Message (0x0)]
	vkEnumeratePhysicalDevices: One or more layers modified physical devices!Count returned by ICDs = 2, count returned above layers = 0
Info
09:52:21
wgpu_hal::vulkan::instance
	objects: (type: INSTANCE, hndl: 0x189d2958ba0, name: ?)
```

**Comment by kvark** [comment]
Interesting. The only layer we are enabling is the validation one, and it wouldn't be as buggy. Maybe something is forcefully injecting a vulkan layer?

**Comment by FredrikNoren** [comment]
@kvark I'm using a fork of wgpu 0.12 with this one commit on top of it that enables `hal::InstanceFlags::DEBUG`  https://github.com/PlayDims/wgpu/commit/e5a578a25a2ba10cd37fb348b074d1e581ab4851, just so I can get more debug information. But this problem was happening before I made the fork.

Tbh I'm not sure if this is a wgpu problem or a driver problem (since reinstalling the drivers makes it work again, even though after a while it stops working for some reason and the drivers have to be reinstalled), but I figured I'd just write what I find here in case there's something that can be done on the wgpu side.
